src/devices/bus/sms_3d/s3dport.h

```diff
@@ -0,0 +1,87 @@
+// license:BSD-3-Clause
+// copyright-holders:Enik Land
+/**********************************************************************
+
+    Input Port for the Sega 3-D Glasses / SegaScope 3-D Glasses
+
+**********************************************************************
+
+
+**********************************************************************/
+
+#ifndef MAME_BUS_SMS_3D_PORT_H
+#define MAME_BUS_SMS_3D_PORT_H
+
+#include "screen.h"
+
+#pragma once
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> sms_3d_port_device
+
+class device_sms_3d_port_interface;
+
+class sms_3d_port_device : public device_t,
+								public device_slot_interface
+{
+public:
+	// construction/destruction
+	template <typename T>
+	sms_3d_port_device(machine_config const &mconfig, char const *tag, device_t *owner, T &&opts, char const *dflt)
+		: sms_3d_port_device(mconfig, tag, owner, 0)
+	{
+		option_reset();
+		opts(*this);
+		set_default_option(dflt);
+		set_fixed(false);
+	}
+
+	sms_3d_port_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	virtual ~sms_3d_port_device();
+
+	DECLARE_WRITE_LINE_MEMBER(write_sscope);
+
+	int m_sscope_state;
+
+	template <typename T> void set_screen_tag(T &&tag) { m_screen.set_tag(std::forward<T>(tag)); }
+
+	required_device<screen_device> m_screen;
```

Comment:
Same comments as the changes to the other slot device.

Reply:
I hope it's done.